templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts

```diff
@@ -9,240 +9,286 @@ import {
+        
+            // Get cognitive models for the current locale.
+            const localizedServices: Partial<ICognitiveModelSet> = this.services.getCognitiveModels();
 
-        const activity: Activity = dc.context.activity;
+            // Run LUIS recognition and store result in turn state.
+            const skillLuis: LuisRecognizer | undefined = localizedServices.luisServices ? localizedServices.luisServices.get("sampleSkill") : undefined;
+            if (skillLuis !== undefined) {
+                const skillResult: RecognizerResult = await skillLuis.recognize(innerDc.context);
+                innerDc.context.turnState.set(StateProperties.skillLuisResult, skillResult);
+            }
+        
+            // Check for any interruptions
+            var interrupted = await this.interruptDialog(innerDc);
+
+            if (interrupted) {
+                // If dialog was interrupted, return EndOfTurn
+                return MainDialog.EndOfTurn;
+            }
+        }
+
+        return await super.onContinueDialog(innerDc);
+    }
+
+    // Runs on every turn of the conversation to check if the conversation should be interrupted.
+    protected async interruptDialog(innerDc: DialogContext): Promise<Boolean> {
+        let interrupted: Boolean = false;
+        const activity: Activity = innerDc.context.activity;
 
         if (activity.type === ActivityTypes.Message && activity.text.trim().length > 0) {
         
             // Get connected LUIS result from turn state.
-            const generalResult: RecognizerResult = dc.context.turnState.get(StateProperties.generalLuisResult);
+            const generalResult: RecognizerResult = innerDc.context.turnState.get(StateProperties.generalLuisResult);
             const intent: string = LuisRecognizer.topIntent(generalResult);
 
-            if(generalResult.intents[intent].score > 0.5) {
+            if (generalResult.intents[intent].score > 0.5) {
                 switch(intent.toString()) {
                     case 'Cancel': { 
 
-                        await dc.context.sendActivity(this.templateEngine.generateActivityForLocale('CancelledMessage'));
-                        await dc.cancelAllDialogs();
-
-                        return InterruptionAction.End;
+                        await innerDc.context.sendActivity(this.templateEngine.generateActivityForLocale('CancelledMessage'));
+                        await innerDc.cancelAllDialogs();
+                        await innerDc.beginDialog(this.initialDialogId);
+                        interrupted = true;
+                        break;
                     } 
                     case 'Help': {
 
-                        await dc.context.sendActivity(this.templateEngine.generateActivityForLocale('HelpMessage'));
-
-                        return InterruptionAction.Resume;
+                        await innerDc.context.sendActivity(this.templateEngine.generateActivityForLocale('HelpMessage'));
```

Comment:
```suggestion
                        await innerDc.context.sendActivity(this.templateEngine.generateActivityForLocale('HelpCard'));
```

Following https://github.com/microsoft/botframework-solutions/blob/v0.8-beta/samples/csharp/skill/SkillSample/Dialogs/MainDialog.cs#L145